base/applications/rapps/gui.cpp

```diff
@@ -257,6 +259,164 @@ class CAppRichEdit:
     }
 };
 
+class CAppInfoDisplay :
+    public CUiWindow<>
+{
+    BOOL bClassRegistered = FALSE;
+
+    LPWSTR pLink = NULL;
+
+public:
+
+    CAppRichEdit * RichEdit;
+
+    BOOL ShowAvailableAppInfo(CAvailableApplicationInfo* Info)
+    {
+        return RichEdit->ShowAvailableAppInfo(Info);
+    }
+
+    BOOL ShowInstalledAppInfo(PINSTALLED_INFO Info)
+    {
+        return RichEdit->ShowInstalledAppInfo(Info);
+    }
+
+    VOID SetWelcomeText()
+    {
+        RichEdit->SetWelcomeText();
+    }
+
+    HWND Create(HWND hwndParent)
+    {
+        if (!bClassRegistered)
+        {
+            WNDCLASSW wndclass = { 0 };
+            wndclass.style = CS_HREDRAW | CS_VREDRAW;
+            wndclass.lpfnWndProc = AppInfoWndProc;
+            wndclass.cbClsExtra = 0;
+            wndclass.cbWndExtra = sizeof(CAppInfoDisplay);
+            wndclass.hInstance = GetModuleHandle(0);
+            wndclass.hIcon = LoadIcon(NULL, IDI_APPLICATION);
+            wndclass.hCursor = LoadCursor(NULL, IDC_ARROW);
+            wndclass.hbrBackground = (HBRUSH)GetStockObject(WHITE_BRUSH);
+            wndclass.lpszMenuName = NULL;
+            wndclass.lpszClassName = L"RAppsAppInfo";
+
+            if (!RegisterClassW(&wndclass)) return 0;
+
+            bClassRegistered = TRUE;
+        }
+
+        m_hWnd = ::CreateWindowExW(0,
+            L"RAppsAppInfo",
+            L"",
+            WS_CHILD | WS_VISIBLE,
+            0, 0, 0, 0,
+            hwndParent, 0, GetModuleHandle(0), this);
+        return m_hWnd;
+    }
+
+    VOID OnLink(ENLINK* Link)
+    {
+        switch (Link->msg)
+        {
+        case WM_LBUTTONUP:
+        case WM_RBUTTONUP:
+        {
+            if (pLink) HeapFree(GetProcessHeap(), 0, pLink);
```

Comment:
This whole `pLink` thing could use some love, I am pretty sure there are some bugs due to that...
(Not something for now, but to keep in mind that if you work on it, evaluate if the whole thing can be made simpler)